arduino/libraries/librariesmanager/install.go

```diff
@@ -85,3 +91,84 @@ func (lm *LibrariesManager) Uninstall(lib *libraries.Library) error {
 	lm.Libraries[lib.Name].Remove(lib)
 	return nil
 }
+
+//InstallZipLib  installs a Zip library on the specified path.
+func (lm *LibrariesManager) InstallZipLib(libPath string) error {
+	libsDir := lm.getUserLibrariesDir()
+	if libsDir == nil {
+		return fmt.Errorf("User directory not set")
+	}
+
+	var filenames []string
+
+	r, err := zip.OpenReader(libPath)
+	if err != nil {
+		return err
+	}
+	defer r.Close()
+
+	for _, f := range r.File {
+
+		// Store filename/path for returning and using later on
+		fpath := filepath.Join(libsDir.String(), f.Name)
+
+		// Check for ZipSlip.
+		if !strings.HasPrefix(fpath, filepath.Clean(libsDir.String())+string(os.PathSeparator)) {
+			return fmt.Errorf("%s: illegal file path", fpath)
+		}
+
+		filenames = append(filenames, fpath)
+
+		if f.FileInfo().IsDir() {
+			// Make Folder
+			os.MkdirAll(fpath, os.ModePerm)
+			continue
+		}
+
+		// Make File
+		if err = os.MkdirAll(filepath.Dir(fpath), os.ModePerm); err != nil {
+			return err
+		}
+
+		outFile, err := os.OpenFile(fpath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, f.Mode())
+		if err != nil {
+			return err
+		}
+
+		rc, err := f.Open()
+		if err != nil {
+			return err
+		}
+
+		_, err = io.Copy(outFile, rc)
+
+		// Close the file without defer to close before next iteration of loop
+		outFile.Close()
+		rc.Close()
+
+		if err != nil {
+			return err
+		}
```

Comment:
Would be great if you could use the [codeclysm/extract](github.com/codeclysm/extract) library, it's already around the CLI, see `arduino/resources/install.go` for example.